src/data/checkfiles/default_checkfiles/unused-metadata.js

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import axios from "axios";
+import { Chip } from "@material-ui/core";
+
+export default {
+  name: "All metadata is used",
+  description:
+    "Checks if every piece of metadata is associated with a user, resource, data object, or collection.",
+  minimum_server_version: "4.2.0",
+  maximum_server_version: "",
+  interval_in_seconds: 604800,
+  active: true,
+  checker: async function () {
+    let result = {
+      status: "",
+      message: "",
+      success: 0,
+      failed: [],
+    };
+
+    let totalMeta = 0 // represents all metadata in the system (not just the ones that are used)
+    let numUnusedMeta = 0; // represents the number of metadata that are not used (not associated with a user, resource, data object, or collection)
+
+    let specificQuery =
+      "SELECT count(unused.*) FROM (select meta_id from R_META_MAIN except select meta_id from R_OBJT_METAMAP) unused";
+
+    let specificWarning = false;
+    const specificResp = await axios({
+      url: `${this.restApiLocation}/query`,
+      method: "GET",
+      headers: {
+        Authorization: localStorage.getItem("zmt-token"),
+      },
+      params: {
+        query: specificQuery,
+        type: "specific",
+        limit: 999999999,
+        offset: 0,
+        "case-sensitive": 1,
+      },
+    }).catch((e) => {
+      specificWarning = true;
+    });
+
+    if (specificResp && specificResp.data) {
+      // specific query succeeded, so we can use the result
+      numUnusedMeta = specificResp.data._embedded[0][0];
+    } else {
+      // specific query failed; first perform a check to see how many total metadata entries there are
+      let totalMetaQuery = "SELECT COUNT(META_DATA_ATTR_ID)";
+
+      const totalMetaResp = await axios({
+        url: `${this.restApiLocation}/query`,
+        method: "GET",
+        headers: {
+          Authorization: localStorage.getItem("zmt-token"),
+        },
+        params: {
+          query: totalMetaQuery,
+          type: "general",
+          limit: 999999999,
+          offset: 0,
+          "case-sensitive": 0,
+        },
+      }).catch(() => {});
+
+      if (totalMetaResp && totalMetaResp.data) {
+        totalMeta = totalMetaResp.data._embedded[0][0];
+      }
+
+      if (totalMeta > 100000) {
```

Comment:
let's pull this to a variable at the top of this checkfile

Reply:
hopefully `maxUnusedMetadataBeforeWarningForNoSpecificQuery` isn't too long of a name 🙃